src/m365/entra/commands/m365group/m365group-set.spec.ts

```diff
@@ -381,6 +392,92 @@ describe(commands.M365GROUP_SET, () => {
     assert(loggerLogToStderrSpy.called);
   });
 
+  it('sets option allowExternalSenders when using delegated permissions', async () => {
+    sinonUtil.restore(accessToken.isAppOnlyAccessToken);
+    sinon.stub(accessToken, 'isAppOnlyAccessToken').returns(false);
+
+    sinon.stub(request, 'patch').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/groups/${groupId}` && opts.data['allowExternalSenders']) {
```

Comment:
Instead of checking data here, let's check it in an assert statement. This will make it a lot easier for us to debug if this test fails.
Let's do this for the other added tests as well.